### PR TITLE
Add an admin-spawn-only syndicate scanner

### PR DESCRIPTION
**Pull request by FunnyMan3595**
## What Does This PR Do
Adds an admin-spawn-only item, the syndicate scanner. On use, this will report the rough range to the nearest mob that:
* has a connected client,
* is not dead,
* is on the same Z level (including inside things),
* and has no special role

If a mob is found, a loud ping noise will be produced.

By default, the ranges are 0-5 tiles, 6-15 tiles, 16-30 tiles, and 30+ tiles (counts as a miss). These ranges and the messages given can be customized in VV.

Can be used once every 10 seconds. This is also configurable in VV.

## Why It's Good For The Game
Mostly useful for an upcoming event by Leanfrog, but could have other uses later.

## Images of changes
image

## Testing
Got Leanfrog to join me, tested ranges.
(added cooldown and sound effects here)
Spawned in as assistant, spawned the scanner. Scanner detected me and made noise.
Made myself a traitor. Scanner detected nobody, was silent.
Tried spamming the scanner. Was told it was recharging.
Reduced the cooldown to 0 in VV. Could spam now.

<hr>

### Declaration
- [X] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/feature_freeze/CODE_OF_CONDUCT.md#currently-changes-to-the-following-types-of-content-reuires-pre-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
(It's a little borderline, but since it needs admin intervention to spawn, I think it's fine.)
<hr>

## Changelog
NPFC

### CODE REVIEW

File: code/game/objects/items/devices/traitordevices.dm [hunk 1]
```diff
@@ -465,3 +465,51 @@
 	GLOB.mirrors -= src
 	QDEL_NULL(appearance_changer_holder)
 	return ..()
+
+/// An admin-spawn item that will tell you roughly how close the nearest loyal Nanotrasen crewmember is.
+/obj/item/syndi_scanner
+	name = "syndicate scanner"
+	desc = "The Syndicate seem to have modified this T-ray scanner to a more nefarious purpose, allowing it to detect all loyal Nanotrasen crew."
```
Comment (Burzah) [inline code/game/objects/items/devices/traitordevices.dm:472]:
```suggestion
	desc = "The Syndicate seem to have modified this T-ray scanner for a more nefarious purpose, allowing it to detect all loyal Nanotrasen crew."
```

File: code/game/objects/items/devices/traitordevices.dm [hunk 2]
```diff
@@ -465,3 +465,51 @@
...
+		var/turf/target_turf = get_turf(player)
+		if(target_turf.z != user_turf.z)
+			continue
+		min_dist = min(min_dist, get_dist(target_turf, user_turf))
+	var/range = 1
+	for(var/test_range in ranges)
+		if(min_dist > test_range)
+			range++
+		else
+			playsound(user, 'sound/effects/ping_hit.ogg', vol = 75, vary = TRUE)
```
Comment (Burzah) [inline code/game/objects/items/devices/traitordevices.dm:513]:
```suggestion
			playsound(user, 'sound/effects/ping_hit.ogg', 75, TRUE)
```
Missed this on my quick glance. More of a nit than anything to keep it inline with the other `playsound` procs used across the repo.

**Comment by Leanfrog** [comment]
Just slapping this on here, if this is going to get closed I'd ask that this PR be kept open for a touch longer as I'd like it TM'd for an event I'm planning, assuming it doesn't get merged.

File: code/game/objects/items/devices/traitordevices.dm [hunk 3]
```diff
@@ -465,3 +465,51 @@
...
+		"Weak signal detected. Range: Within 30 meters.",
+		"No signal detected."
+	)
+	var/last_use = 0
+	var/cooldown = 10 SECONDS
+
+/obj/item/syndi_scanner/attack_self(mob/user)
+	if(last_use > world.time - cooldown)
+		to_chat(user, "<span class='warning'>[src] is recharging!</span>")
+		return
```
Comment (DGamerL) [inline code/game/objects/items/devices/traitordevices.dm:494]: Change this all with TG cooldowns please

**Review by Sirryan2002 — APPROVED** [review]

**Review by Burzah — APPROVED** [review]

File: code/game/objects/items/devices/traitordevices.dm [hunk 4]
```diff
@@ -465,3 +465,60 @@
...
+	COOLDOWN_DECLARE(scan_cooldown)
+	var/on_hit_sound = 'sound/effects/ping_hit.ogg'
+
+/obj/item/syndi_scanner/attack_self(mob/user)
+	if(!COOLDOWN_FINISHED(src, scan_cooldown))
+		to_chat(user, "<span class='warning'>[src] is recharging!</span>")
+		return
+
+	COOLDOWN_START(src, scan_cooldown, cooldown_length)
+	last_use = world.time
```
Comment (DGamerL) [inline code/game/objects/items/devices/traitordevices.dm:501]:
```suggestion
```

File: code/game/objects/items/devices/traitordevices.dm [hunk 5]
```diff
@@ -465,3 +465,60 @@
...
+	/// Split points for range_messages.
+	var/list/ranges = list(5, 15, 30)
+	/// Messages to output to the user.
+	var/list/range_messages = list(
+		"Very strong signal detected. Range: Within 5 meters.",
+		"Strong signal detected. Range: Within 15 meters.",
+		"Weak signal detected. Range: Within 30 meters.",
+		"No signal detected."
+	)
+	var/last_use = 0
```
Comment (DGamerL) [inline code/game/objects/items/devices/traitordevices.dm:490]:
```suggestion
```

**Review by DGamerL — APPROVED** [review]
Unnecessary var detected!

**Review by DGamerL — APPROVED** [review]

**Pull request merged**